kolibri/core/discovery/test/test_api.py

```diff
@@ -102,3 +104,56 @@ def test_reading_network_location_list_filter_soud(self):
         )
         for location in response.data:
             self.assertFalse(location["subset_of_users_device"])
+
+
+class PinnedDeviceAPITestCase(APITestCase):
+    @classmethod
+    def setUpTestData(cls):
+        provision_device()
+        cls.facility = FacilityFactory.create()
+        cls.user = FacilityUserFactory(facility=cls.facility)
+        cls.network_location = models.NetworkLocation.objects.create(
+            base_url="https://kolibrihappyurl.qqq/"
+        )
+        cls.network_location2 = models.NetworkLocation.objects.create(
+            base_url="https://anotherone.moc"
+        )
+
+    def setUp(self):
+        self.client.login(
+            username=self.user.username, password=DUMMY_PASSWORD, facility=self.facility
+        )
+
+    def test_add_pinned_device(self):
+        """
+        Tests the API for adding Pinned Devices
+        """
+        response = self.client.post(
+            reverse("kolibri:core:pinned_devices-list"),
+            data={"user": self.user.id, "device_id": self.network_location.id},
```

Comment:
Everything works as expected for me and no errors in the console  🎉 

Not sure if this has already been caught, but since `device_id` has been updated to `instance_id`, the remaining references to `device_id` in this file need to be updated as well.